Move Children to Builder

pzalep1 opened the pull request:
The purpose of this PR is to move the adding and deleting children functionality into the builder and out of the dashboard. This will allow for us to redesign the dashboard without children. 
<img width="555" alt="Screen Shot 2019-06-29 at 9 56 12 AM" src="https://user-images.githubusercontent.com/43140629/60385157-262e2700-9a54-11e9-999d-0aa239efd5a1.png">
<img width="615" alt="Screen Shot 2019-06-29 at 9 55 11 AM" src="https://user-images.githubusercontent.com/43140629/60385160-2cbc9e80-9a54-11e9-9693-e83767dbd072.png">
<img width="977" alt="Screen Shot 2019-06-29 at 9 55 18 AM" src="https://user-images.githubusercontent.com/43140629/60385163-3514d980-9a54-11e9-831f-4b3f90700baf.png">
<img width="761" alt="Screen Shot 2019-06-29 at 9 55 40 AM" src="https://user-images.githubusercontent.com/43140629/60385167-3940f700-9a54-11e9-9b1e-c3a5bd392732.png">




Also, add a message for Nanomodules: 
<img width="340" alt="Screen Shot 2019-06-24 at 2 59 11 PM" src="https://user-images.githubusercontent.com/43140629/60044565-a89f9b00-9690-11e9-8f5c-b291139914f7.png">

**Note: I am still finishing up the toggle component.

nrwinner commented on src/app/onion/learning-object-builder/components/scaffold/scaffold.component.html:
```diff
@@ -19,23 +24,57 @@
+    <div class="modal-title">Just to confirm...</div>
+      <div class="modal-text">{{ childrenConfirmationMessage }}</div>
+      <div class="btn-group center">
+        <button class="button good" (click)="deleteChild(i)">Yep, do it! <i class="far fa-check"></i></button>
```
add an aria label here pls

nrwinner commented on src/app/onion/learning-object-builder/components/scaffold/scaffold.component.html:
```diff
@@ -19,23 +24,57 @@
+      <div class="modal-text">{{ childrenConfirmationMessage }}</div>
+      <div class="btn-group center">
+        <button class="button good" (click)="deleteChild(i)">Yep, do it! <i class="far fa-check"></i></button>
+        <button class="button neutral" (click)="toggleConfirmationModal(false)">Wait, nevermind! <i class="far fa-ban"></i></button>
```
add an aria label here pls

pzalep1 replied:
Resolved with d52869d

nrwinner commented on src/app/onion/learning-object-builder/components/scaffold/add-child/add-child.component.html:
```diff
@@ -0,0 +1,40 @@
+<div class="search-bar">
+  <input aria-label="Search for Children" type="search" placeholder="Search for Children" [(ngModel)]="childrenSearchString" (ngModelChange)="searchString$.next($event)"/>
+  <button class="search-button" (click)="search()">
```
add an aria label here pls

pzalep1 replied:
Resolved with d52869d

nrwinner commented on src/app/onion/learning-object-builder/components/scaffold/add-child/add-child.component.html:
```diff
@@ -0,0 +1,40 @@
+            <div class = "child-last-updated">{{c.date |  date:'shortDate'}}</div>
+          </div>
+        </div>
+        <button class="add" (click)="addChildToList(i)">
```
add an aria label here pls

pzalep1 replied:
**Resolved with d52869d**

nrwinner commented on src/app/onion/learning-object-builder/components/scaffold/scaffold.component.html:
```diff
@@ -19,23 +24,57 @@
+      <div [ngStyle]="{display: editContent ? 'none' : 'block'}" class="hamburger-handle" cdkDragHandle>
         <i class="fas fa-bars"></i>
       </div>
+      <button [ngStyle]="{display: editContent ? 'block' : 'none'}" class="delete" (click)="deleteButton(i)">
```
add an aria label here pls

pzalep1 replied:
Resolved with d52869d

DonSeannelly commented:
We should make sure that we validate the UI being usable for users who have many Learning Objects in their dashboards. Not necessarily to Dr. Kaza's size, but perhaps a user who has composed one or two courses out of micro or nanomodules.

pzalep1 commented:
> We should make sure that we validate the UI being usable for users who have many Learning Objects in their dashboards. Not necessarily to Dr. Kaza's size, but perhaps a user who has composed one or two courses out of micro or nanomodules.

@DonSeannelly I updated the screenshots. We moved the button to the top so that if the object has 14 or 15 micro or nanomodules the add button comes up at the top and then new child gets appended to the front of the list instead of the back.

pzalep1 commented on src/app/onion/learning-object-builder/components/scaffold/scaffold.component.html:
```diff
@@ -19,23 +24,57 @@
 </ng-template>
+
+<ng-template #addChild>
+  <button #addChildButton class="edit-children" (click)="toggleAddChild(); $event.stopPropagation()" *ngIf="learningObject?.length !== 'nanomodule'">
```
Resolved with d52869d

nrwinner commented on src/app/onion/learning-object-builder/components/scaffold/scaffold.component.scss:
```diff
@@ -20,8 +20,7 @@
   .edit {
-    margin-left: auto;
-    padding-right: 50px;
+    margin-left: 12%;
```
Why use a percentage value here for the margin?
